client/python/gradio_client/utils.py

```diff
@@ -383,8 +383,8 @@ def decode_base64_to_file(
     dir: str | Path | None = None,
     prefix: str | None = None,
 ):
-    if dir is not None:
-        os.makedirs(dir, exist_ok=True)
+    directory = Path(dir or tempfile.gettempdir()) / secrets.token_hex(20)
```

Comment:
I'm not sure we need the `secrets.token_hex(20)` here since the use of tempfile should already create a unique name.

Reply:
You're right, but I did it this way just to have consistent behavior for all of the components.